src/awkward/_util.py

```diff
@@ -536,1377 +801,389 @@ def extra(args, kwargs, defaults):
     return out
 
 
-def key2index(keys, key):
-    if keys is None:
-        attempt = None
-    else:
-        try:
-            attempt = keys.index(key)
-        except ValueError:
-            attempt = None
-
-    if attempt is None:
-        m = key2index._pattern.match(key)
-        if m is not None:
-            attempt = m.group(0)
-
-    if attempt is None:
-        raise ValueError(
-            f"key {key!r} not found in record" + exception_suffix(__file__)
-        )
-    else:
-        return attempt
-
-
-key2index._pattern = re.compile(r"^[1-9][0-9]*$")
+# def key2index(keys, key):
+#     if keys is None:
+#         attempt = None
+#     else:
+#         try:
+#             attempt = keys.index(key)
+#         except ValueError:
+#             attempt = None
 
+#     if attempt is None:
+#         m = key2index._pattern.match(key)
+#         if m is not None:
+#             attempt = m.group(0)
 
-def completely_flatten(array):
-    if isinstance(array, ak.partition.PartitionedArray):
-        out = []
-        for partition in array.partitions:
-            for outi in completely_flatten(partition):
-                out.append(outi)
-        return tuple(out)
+#     if attempt is None:
+#         raise error(ValueError(
+#             "key {0} not found in record".format(repr(key))
+#         ))
+#     else:
+#         return attempt
 
-    elif isinstance(array, virtualtypes):
-        return completely_flatten(array.array)
 
-    elif isinstance(array, unknowntypes):
-        return (ak.nplike.of(array).array([], dtype=np.bool_),)
+# key2index._pattern = re.compile(r"^[1-9][0-9]*$")
 
-    elif isinstance(array, indexedtypes):
-        return completely_flatten(array.project())
 
-    elif isinstance(array, uniontypes):
-        out = []
-        for i in range(array.numcontents):
-            tmp = completely_flatten(array.project(i))
-            assert isinstance(tmp, tuple)
-            for x in tmp:
-                out.append(x)
-        return tuple(out)
+# def make_union(tags, index, contents, identifier, parameters):
```

Comment:
The same is true of `make_union`: we never needed it in v2 and it can be deleted.